File: hexchat/autorejoin.pl
use strict;

Xchat::register("Autorejoin", "1.0");
Xchat::print("Autorejoin script loaded\n");

Xchat::hook_print('You Kicked', sub {
	Xchat::hook_timer(1500, sub {
		Xchat::command("join $_[0]");
		return Xchat::REMOVE;
	}, $_[0][1]);

	return Xchat::EAT_NONE;
});

File: hexchat/autorejoin.py
import hexchat

__module_name__ = 'AutoRejoin'
__module_version__ = '1.0'
__module_description__ = ''

hexchat.prnt('Autorejoin script loaded')

def autorejoin_callback(word, word_eol, user_data):
	hexchat.hook_timer(3000, rejoin_timer_callback, word[1])

def rejoin_timer_callback(user_data):
	hexchat.command('join {0}'.format(user_data))

hexchat.hook_print('You Kicked', autorejoin_callback)
